internal/dumper/pool.go

```diff
@@ -40,7 +40,7 @@ func (conn *Connection) StreamFetch(query string) (driver.Rows, error) {
 // NewPool creates the new pool.
 func NewPool(log *zap.Logger, cap int, address string, user string, password string, vars []string, database string) (*Pool, error) {
 	conns := make(chan *Connection, cap)
-	for i := 0; i < cap; i++ {
+	for i := 1; i <= cap; i++ {
```

Comment:
Any reason for this change? Starting from `0` and using `<` is the canonical style.

Reply:
I've reverted this back to the canonical style in NewPool.

The only remaining question would be whether I should leave things as-is for `--show-details` with the zero-based output like this:

```
Started Processing Data File: metrics.connections.00001.sql in thread 2 (File 1 of 5)
Started Processing Data File: metrics.connections.00004.sql in thread 1 (File 4 of 5)
Started Processing Data File: metrics.connections.00002.sql in thread 3 (File 2 of 5)
Started Processing Data File: metrics.connections.00003.sql in thread 0 (File 3 of 5)
```

Or whether I should additionally tweak the output so it appears as 1-based.

If it's fine as-is then no additional updates should be needed.